src/Graph/Provider.php

```diff
@@ -102,7 +102,7 @@ protected function mapUserToObject(array $user)
         return (new User())->setRaw($user)->map([
             'id'    => $user['id'],
             'name'  => $user['displayName'],
-            'email' => $user['mail'],
+            'email' => $user['userPrincipalName'],
```

Comment:
```suggestion
            'email' => $user['mail'] ?? $user['userPrincipalName'],
```

How about we check both? That way no behavior should change and it also fixes your case.